directory/datadog_checks/directory/directory.py

```diff
@@ -48,8 +50,13 @@ def check(self, instance):
         filetagname = instance.get("filetagname", "filename")
         filegauges = _is_affirmative(instance.get("filegauges", False))
         countonly = _is_affirmative(instance.get("countonly", False))
+        ignore_missing = _is_affirmative(instance.get("ignore_missing", False))
 
         if not exists(abs_directory):
+            if ignore_missing:
+                self.log.debug("DirectoryCheck: the directory (%s) does not exist.  Skipping." % abs_directory)
+                return
```

Comment:
I dunno if I like it returning nothing, but I also can't think of something else for it to do that would be good. So, it's good enough for now.